src/object_storage/objects.rs

```diff
@@ -145,6 +145,11 @@ impl Object {
         name: ref String
     }
 
+    transparent_property! {
+        #[doc = "Object hash or ETag."]
+        hash: ref String
```

Comment:
nit: I'd prefer properties to be sorted, i.e. hash should go before name.

Actually, it seems that it should be `Option<String>` - see my other comments.

Reply:
I re-order properties.